rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -49,28 +71,113 @@
     private FlowNodeUtil() {
     }
 
-    public static long getNodeExecDuration(FlowNode node) {
-        List<FlowNode> childNodes = getChildNodes(node);
-        if (!childNodes.isEmpty()) {
-            long startTime = TimingAction.getStartTime(node);
-            long endTime = TimingAction.getStartTime(childNodes.get(childNodes.size() - 1));
+    public abstract static class CacheExtensionPoint implements ExtensionPoint {
```

Comment:
This whole implementation is iffy to me.  We're creating an ExtensionPoint & Extension solely to hack around the fact that the JenkinsRule does classloading wrong (the same classloader is used across tests).  The initial implementation with the static cache + invalidation was simpler and more straightforward.  But I raised my protest about that already and was overruled, so if it's going to be ugly, at least it's be confined to a few parts of one file. 

I think that there are far better ways to do this overall, but rather than try to put a dress and bowties on a pig, I'm going to save it for architectural changes along with the new FlowAnalyzer (hopefully I can find a place where it's logical and easy to put the cache, rather than in the Util class).